lib/statsd.js

```diff
@@ -346,6 +345,11 @@ Client.prototype.sendMessage = function (message, callback) {
     const errFormatted = err ? new Error(`Error sending hot-shots message: ${err}`) : null;
     if (errFormatted) {
       errFormatted.code = err.code;
+      // handle UDS error that requires socket replacement when we are not
+      // emitting the `error` event on `this.socket`
+      if (this.protocol === PROTOCOL.UDS && (callback || this.errorHandler)) {
+        udsErrorHandler(this, err);
+      }
```

Comment:
If I understand correctly this is because the code below does the following:

```js
if (callback) {
  callback(errFormatted, bytes);
} else if (err && this.errorHandler) {
  this.errorHandler(errFormatted);
} else if (err) {
  console.error(String(errFormatted));
  this.socket.emit('error', errFormatted);
}
```

i.e. the only case that the `error` event is emitted (when `err` is truth-y) is when `callback` and `errorHandler` are false-y?

Does it not make sense (it's a little repetitive) to `this.socket.emit()` in all cases?

Reply:
Exactly. I tried emitting the error on the socket in all cases at first but decided against it because of this comment:
https://github.com/chotiwat/hot-shots/blob/450715ecc308addd74640b514e2a78330a194a48/lib/statsd.js#L361-L362